### PR TITLE
Confirm user api v2 endpoint

### PR DESCRIPTION
**Pull request by pmilewski**

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -49,6 +49,7 @@
     namespace :v2 do
       namespace :storefront do
         resource :account, controller: :account, only: %i[show create update]
+        resource :confirmation, controller: :confirmations, only: %i[show]
```
Comment (damianlegawiec) [inline config/routes.rb:52]:
```suggestion
        resource :confirmations, only: %i[show]
```

File: app/controllers/spree/api/v2/storefront/confirmations_controller.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+module Spree
+  module Api
+    module V2
+      module Storefront
+        class ConfirmationsController < ::Spree::Api::V2::BaseController
+
+          def show
+            user = Spree.user_class.confirm_by_token(params[:confirmation_token])
```
Comment (damianlegawiec) [inline app/controllers/spree/api/v2/storefront/confirmations_controller.rb:8]: @pmilewski will it return a 404 when not found?

File: app/controllers/spree/api/v2/storefront/confirmations_controller.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+module Spree
+  module Api
+    module V2
+      module Storefront
+        class ConfirmationsController < ::Spree::Api::V2::BaseController
```
Comment (damianlegawiec) [inline app/controllers/spree/api/v2/storefront/confirmations_controller.rb:5]:
```suggestion
        class AccountConfirmationsController < ::Spree::Api::V2::BaseController
```

File: app/controllers/spree/api/v2/storefront/confirmations_controller.rb (hunk 2 above)
Comment (pmilewski) [reply app/controllers/spree/api/v2/storefront/confirmations_controller.rb:8]: 422: Confirmation token is invalid

File: config/routes.rb [hunk 4]
```diff
@@ -49,6 +49,7 @@
     namespace :v2 do
       namespace :storefront do
         resource :account, controller: :account, only: %i[show create update]
+        resource :account_confirmations, only: %i[show]
```
Comment (damianlegawiec) [inline config/routes.rb:52]:
```suggestion
        resources :account_confirmations, only: %i[show]
```

File: app/controllers/spree/api/v2/storefront/account_confirmations_controller.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  module Api
+    module V2
+      module Storefront
+        class AccountConfirmationsController < ::Spree::Api::V2::BaseController
+
+          def show
+            user = Spree.user_class.confirm_by_token(params[:id])
+
+            if user.errors.empty?
+              head :ok
```
Comment (tniezg) [inline app/controllers/spree/api/v2/storefront/account_confirmations_controller.rb:11]: We should know if the user needs to wait for manual account approval or can log in immediately. For example, with a response like `{"status": "active"}` and `{"status": "manual"}`. @pmilewski @damianlegawiec

Comment (pmilewski) [reply app/controllers/spree/api/v2/storefront/account_confirmations_controller.rb]: I added a response with the state.

File: spec/requests/spree/api/v2/storefront/account_confirmation_spec.rb [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+describe 'Storefront API v2 Account Confirmation spec', type: :request do
+  describe 'account_confirmations#show' do
+
+    before do
+      set_confirmable_option(true)
+      get "/api/v2/storefront/account_confirmations/#{confirmation_token}"
+    end
+
+    after(:each) { set_confirmable_option(false) }
+
```
Comment (pmilewski) [inline spec/requests/spree/api/v2/storefront/account_confirmation_spec.rb:12]: I am still trying to fix failing tests.
It seems like the above blocks have an impact on other spec files.

**Pull request merged**